main.go

```diff
@@ -46,6 +54,13 @@ var (
 	setupLog = ctrl.Log.WithName("setup")
 )
 
+type hasher struct{}
+
+func (h hasher) Sum64(data []byte) uint64 {
+	// TODO: Investigate hash function options.
+	return xxhash.Sum64(data)
```

Comment:
In initial tests, this seems to be working. If the metrics turn out to be very unbalanced on large numbers we can revisit uniform distribution hash options